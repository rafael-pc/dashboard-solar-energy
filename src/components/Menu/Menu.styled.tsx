import styled from "styled-components";
import { theme } from "../../styles";

export const Container = styled.div`
  @media only screen and (min-width: ${theme.screen.mobile}) {
    #menu-mobile {
      display: none;
    }
  }
`;
