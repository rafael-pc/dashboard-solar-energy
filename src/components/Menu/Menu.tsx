import React, { useState, useRef } from "react";
import { Container } from "./Menu.styled";
import { Burger, MenuMobile } from "..";
import FocusLock from "react-focus-lock";

const Menu: React.FC = () => {
  const [open, setOpen] = useState(false);
  const node = useRef<HTMLDivElement>(null);
  const menuId = "main-menu";

  return (
    <Container>
      <div ref={node} id="menu-mobile">
        <FocusLock disabled={!open}>
          <Burger open={open} setOpen={setOpen} aria-controls={menuId} />
          <MenuMobile open={open} setOpen={setOpen} id={menuId} />
        </FocusLock>
      </div>
    </Container>
  );
}

export default Menu;

