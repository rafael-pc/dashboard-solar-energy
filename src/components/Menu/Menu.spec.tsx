import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Menu from './Menu';
import "@testing-library/jest-dom";

describe('Menu', () => {
  it('renders correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Menu />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
