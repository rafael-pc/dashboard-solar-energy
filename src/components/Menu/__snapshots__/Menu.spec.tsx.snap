// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
<div>
  <div
    class="sc-bcPKhP gmgKME"
  >
    <div
      id="menu-mobile"
    >
      <div
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="-1"
      />
      <div
        data-focus-lock-disabled="disabled"
      >
        <button
          aria-controls="main-menu"
          aria-expanded="false"
          aria-label="Toggle menu"
          class="sc-grXZZQ IkEIg"
        >
          <span />
          <span />
          <span />
        </button>
        <nav
          aria-hidden="true"
          class="sc-ipMvLY fQGCDQ"
          id="main-menu"
        >
          <a
            href="/dashboard"
            tabindex="-1"
          >
            Dashboard
          </a>
          <a
            href="/units"
            tabindex="-1"
          >
            Unidade consumidora
          </a>
          <a
            href="/register_month"
            tabindex="-1"
          >
            Cadastro de en√©rgia gerada
          </a>
        </nav>
      </div>
      <div
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;
